### DISCUSSION TITLE
Alpine static builds - no output from iperf3 -h

### DISCUSSION THREAD

**Discussion by userdocs** [esnet/iperf#1237] [Q&A] [answered]
I am building iperf3 from GitHub using Alpine linux and docker multiarch

https://github.com/userdocs/iperf3-static/releases/tag/3.10.1+

For some reason when try to use the `-h` help flag i get no output. are the docs linked and not built into the bin or is it something else?

```
wget -qO ~/iperf3 https://github.com/userdocs/iperf3-static/releases/latest/download/iperf3-amd64
chmod 700 ~/iperf3
```

Version works fine

```
~/iperf3 -v
```

Help has no output

```
~/iperf3 -h
```

**Comment by bmah888** [comment]
`iperf3 -h` should always return output, whether the binary is statically or dynamically linked. The help text is compiled into the binary (and we can see that by visual inspection of the binary file). I built an iperf3 binary that was configured with `configure --disable-shared --enable-static-bin` and `iperf3 -h` behaved sanely. I can't think of a reason that this wouldn't work...

**Reply by userdocs** [reply]
Any chance you could try on Alpine linux (musl based) and see if it happens? I have the info required here

https://github.com/userdocs/iperf3-static#generic-build-dependencies

**Reply by bmah888** [reply]
Well...OK I threw together an Alpine VM and your binary and sure enough `iperf3 -h` doesn't return any output. It did seem to be able to run tests however. Also I did `apk iperf3` and the dynamically linked executable that got installed did have a working help output.

So I confirm the behavior you saw, but I have no idea what's causing it. Is this the only problem you're seeing?

**Reply by userdocs** [reply]
I have not encountered any bugs or noticed any, It was just the help thing so far.

**Comment by davidBar-On** [comment]
`iperf3 -h` is done using `fprintf(stdout, ...)`, while `'iperf3 -v` is done using `printf(...)`.

It may be that Alpine Linux does not flush the output buffers before the program ends, so `fflush(...)` should be added in `usage()`, `usage_long()` and `warning()`. (I can submit a PR with these changes if it is easier.)

**Comment by userdocs** [comment, marked as answer]
@bmah888 and @davidBar-On i think this pull has fixed it.

https://github.com/esnet/iperf/pull/1330

It works normally now in my builds https://github.com/userdocs/iperf3-static/releases/tag/3.11
